taskcat/testing/manager.py

```diff
@@ -181,7 +178,7 @@ def end(
             self.test_definition.delete_stacks()
 
         if not dont_wait_for_delete:
-            terminal_printer.report_test_progress(stacker=self.test_definition)
+            self.printer.report_test_progress(stacker=self.test_definition)
```

Comment:
I don't think we can totally separate this because of how the terminal printer works. So all printer objects will need to block when this method is called. The good thing is we are already passing in the stacker object and once the stacker.status is recursive it should be easy to default the abstract base printer to block when called until the stack is finished.